src/list/state.rs

```diff
@@ -345,6 +347,39 @@ impl<'a> ListState<'a> {
         Ok(())
     }
 
+    pub fn select_if_matches_search_query(&mut self) {
+        let idx = self
+            .app_state
+            .exercises()
+            .iter()
+            .filter(|exercise| match self.filter() {
+                Filter::None => true,
+                Filter::Done => exercise.done,
+                Filter::Pending => !exercise.done,
+            })
+            .enumerate()
+            .find_map(|(i, s)| {
+                if s.name.contains(self.search_query.as_str()) {
+                    Some(i)
+                } else {
+                    None
+                }
+            });
```

Comment:
Please use [`position`](https://doc.rust-lang.org/stable/std/iter/trait.Iterator.html#method.position) instead of `enumerate` with `find_map`.

I also try to avoid such short variable names. Better use `exercise` instead of `s`.

Reply:
made the refactor, out of curiosity, is there a reason why position() is preferrable?

Reply:
Shorter and easier to understand what one is computing.